[Feat] 모달 (새 파일 생성) & 모달 (프로필 설정) 제작 및 의존성을 가진 컴포넌트 리팩토링

kyeoungwoon opened the pull request:
# Draft라 리뷰 중간에 해주셔도 되는데 변경될게 조금 있긴 합니다!

## #️⃣ Related Issues

> 연관된 모든 Issue를 작성해주세요.

- Issue #96 #154 
- Issue #97

## 🧑‍💻 작업 내용

> 어떤 작업을 진행했는지 자세하게 작성해주세요.

- DatePicker의 props 중 초기값을 전달해주는 부분을 삭제했습니다. (필요없음 + 불필요한 리렌더링 발생)
- TimePicker에서 비활성화를 할 수 있도록 했습니다.

- 새로운 회의록 모달
- 새로운 이벤트 모달
- 새로운 팀 분위기 설문 모달 
- 프로필 설정 모달 

커밋 내역이 작업내역과 일치합니다. 확인 부탁드립니다.

## 💾 작업 결과 (선택)

> 사진, 영상 등을 첨부해주세요.

## 💬 리뷰 시 요청사항 (선택)

> Reviewer는 코드 리뷰의 코멘트에 코멘트를 강조하고 싶은 정도를 [Pn 규칙](https://blog.banksalad.com/tech/banksalad-code-review-culture/#%EC%BB%A4%EB%AE%A4%EB%8B%88%EC%BC%80%EC%9D%B4%EC%85%98-%EB%B9%84%EC%9A%A9%EC%9D%84-%EC%A4%84%EC%9D%B4%EA%B8%B0-%EC%9C%84%ED%95%9C-pn-%EB%A3%B0)에
> 맞춰서 표기해 주세요.

-

## 📝 Checklist

- [ ] Reviewer를 추가했나요?
- [ ] Convention을 준수했나요?

kyeoungwoon committed:
✨ Feat: on/off 토글 버튼 제작 및 shadows.css 파일 병합

파일 병합은 단순히 구분된 layer를 합치기만 했습니다.

kyeoungwoon committed:
✨ Feat: 새로운 회의록 추가 모달(새 파일 생성) 추가

File을 부모에서 관리하는 방식에 따라 수정을 요합니다

kyeoungwoon committed:
📝 Docs: 새로운 회의록 모달에 JSDoc 추가

raw 한 버전입니다. 참고용!

kyeoungwoon committed:
✨ Feat: change toggle button to cursor pointer

kyeoungwoon committed:
🔀 Merge: develop

kyeoungwoon committed:
✨ Feat: 업로드할 수 있는 파일 형식 제한 추가

kyeoungwoon committed:
🔧 Fix: set max width in file name field

디자이너님 요청 사항

kyeoungwoon committed:
🔀 Merge: update with develop

kyeoungwoon committed:
📦️ Package: 불필요한 패키지 삭제 및 devDependencies로 이동

kyeoungwoon committed:
✨ Feat: InputFieldModal 컴포넌트 분리

공통된 컴포넌트는 분리 처리합니다

kyeoungwoon committed:
♻️ Refactor: space-*를 gap-*로 변경

kyeoungwoon committed:
🔧 Fix: DatePicker Header에서 button 안에 button 있는 문제 해결

kyeoungwoon committed:
🔧 Fix: DatePicker 선택 안되는 문제 해결 (리렌더링)

build 전 테스트 커밋

kyeoungwoon committed:
♻️ Refactor: 토글 버튼 동작 수정

kyeoungwoon committed:
♻️ Refactor: build 시 unused_var 문제 해결

workspace 생성 부분에서 isOwner는 추후 확장성 고려하여 냅뒀습니다.

kyeoungwoon committed:
♻️ Refactor: 태그 선택 시 border 떄문에 매번 바뀌던 문제 임시 해결 (resolves #150)

kyeoungwoon committed:
✨ Feat: 이벤트 생성 모달 1차 제작 (미완성)

kyeoungwoon committed:
♻️ Refactor: DatePicker 초기화 시에 props로 default date 넘길 수 없도록 변경

kyeoungwoon committed:
🔧 Fix: DatePicker 컴포넌트 변경에 따른 스토리북 변경

kyeoungwoon committed:
♻️ Refactor: DateTimePicker 컴포넌트화해서 분리

추후 class는 동적으로 설정 가능하게 해도 좋을 것 같음

kyeoungwoon committed:
🩹 Chore: test page 수정

kyeoungwoon committed:
♻️ Refactor: InputFieldModal에 title을 optional로 변경, className 추가

kyeoungwoon committed:
✨ Feat: 새로운 이벤트 생성 모달 제작 Step 3/4 (키워드 로직 제외)

kyeoungwoon committed:
✨ Feat: Chip 방식의 Input Field 컴포넌트화

이메일 초대에서 변형하였습니다

kyeoungwoon committed:
🔧 Fix: 불필요하게 들어가있던 tailwindcss 클래스 제거

kyeoungwoon committed:
✨ Feat: Chip 방식의 Keyword Input 구성

duwlsssss commented on src/common/components/DatePicker/DatePicker.client.tsx:
```diff
@@ -14,43 +14,18 @@ import DatePickerHeader from './DatePickerHeader/DatePickerHeader.client';
-      setSelected(newSelected);
-    }
-  }, [selected, selectedDates]);
+  // 외부에서 props로 selectedDates를 전달했을 때, 업데이트 하기 위한 로직입니다.
+  // selectedDates를 삭제 처리하였습니다. (부모에서 전달할 수 없습니다.)
```
P3: 여기 주석은 다들 확인하면 지워도 될 것 같아요!

kyeoungwoon replied:
약간 버저닝 느낌으로 나중에 초기값 세팅 다시 하는 것 처럼 번복할까봐 남겨둔거긴 해요!

kyeoungwoon replied:
P5: 여기는 우선 다른 사람이 사용 시에 헷갈릴 수도 있을 것 같아서 히스토리 차원에서 남겨두도록 하곘습니다.

duwlsssss commented on src/common/components/DatePicker/DatePicker.stories.tsx:
```diff
@@ -29,12 +26,11 @@ type Story = StoryObj<DatePickerProps>;
 // Interactive wrapper to manage state internally
 const InteractiveWrapper = (args: DatePickerProps) => {
-  const [selectedDates, setSelectedDates] = useState<Date[]>(args.selectedDates || []);
+  // selectedDates prop 제거
```
P3: 여기도 없애도 되겠네요

kyeoungwoon replied:
스토리북은 GPT의 힘으로 제작하여서 .. 헛허 수정하겠습니다

kyeoungwoon replied:
P5: 지금 보니까 주석 삭제를 요구하신 것 같네요. 반영하였습니다.

duwlsssss commented on src/common/components/buttons/30px/DownloadButton.tsx:
```diff
@@ -7,7 +7,7 @@ const DownloadButton = () => {
     <button
       className={clsx(
-        'text-button-3 inline-flex h-[30px] w-[96px] items-center justify-center space-x-[4px] rounded-[100px] bg-gray-100 py-[6px] pr-[14px] pl-[12px] text-white',
+        'text-button-3 inline-flex h-[30px] w-[96px] items-center justify-center gap-x-[4px] rounded-[100px] bg-gray-100 py-[6px] pr-[14px] pl-[12px] text-white',
```
P5: 버튼은 대대적인 pxr 변경 작업 필요하겠다...

kyeoungwoon replied:
나중에 한번에 슥삭 바꿔둘께여

kyeoungwoon replied:
해당 버튼에서는 반영해두었습니다.

duwlsssss commented on src/common/components/etc/BoxedFile/BoxedFile.client.tsx:
```diff
@@ -39,7 +41,7 @@ const BoxedFile = ({
-          <img
+          <Image
             src={thumbnailUrl}
             alt=""
```
P2: 대체 텍스트로 "파일 이미지" 정도 넣어놓는 건 어떨까요?

kyeoungwoon replied:
Build 할 때 warning 뜨는 거 떄문에 임시로 바꿔둔거긴 한데 좋습니다!

kyeoungwoon replied:
반영하였습니다.

duwlsssss commented on src/common/components/inputs/modals/InputFieldModal/InputFieldModal.client.tsx:
```diff
@@ -0,0 +1,29 @@
+    <div className={clsx('w-534pxr gap-y-8pxr flex flex-col justify-center', className)}>
+      {title && <p className="text-cap1-rg h-16pxr text-left text-gray-200">{title}</p>}
+      <input
+        type={type || 'text'}
```
P3: type이 null || undefined일 경우만 고려하려면 `??` (null 병합 연산자) 쓰는 게 더 좋겠어용

kyeoungwoon replied:
`??`의 존재 여부를 가끔씩 까먹네요. 감사합니다.
`||`을 사용할 경우에, `type = ""`일 경우에 기본값이 아닌 공백문자열을 그대로 반환한다고 하네요.

kyeoungwoon replied:
반영하였습니다.

duwlsssss commented on src/common/components/modals/CommonText/CommonText.server.tsx:
```diff
@@ -0,0 +1,22 @@
+import clsx from 'clsx';
+
+import { CommonTextProps, CommonTextType } from './CommonText.types';
+
+const CommonText = ({ text, type, className }: CommonTextProps) => {
+  switch (type) {
+    case CommonTextType.T2_BD:
+      return <p className={clsx('text-t2-bd text-black', className)}>{text}</p>;
+    case CommonTextType.T3_BD:
+      return <p className={clsx('text-t3-bd text-black', className)}>{text}</p>;
+    case CommonTextType.T5_SB:
+      return <p className={clsx('text-t5-sb text-black', className)}>{text}</p>;
+    case CommonTextType.T6_SB:
+      return <p className={clsx('text-t6-sb text-black', className)}>{text}</p>;
+    case CommonTextType.CAP1_RG:
+      return <p className={clsx('text-cap1-rg text-gray-300', className)}>{text}</p>;
+    default:
+      return <p className={clsx('', className)}>ERR_CHCK_TYPE</p>; // Fallback for any other type
+  }
+};
+
+export default CommonText;
```
P4: 오 이렇게 나눈 이유는 뭔가요??

kyeoungwoon replied:
모달 내에서 계속 중복되서 사용되길래 귀찮아서 넣었습니다 .....

duwlsssss commented on src/common/hooks/useCreateEventConditions.ts:
```diff
@@ -0,0 +1,157 @@
+  },
+  friendTag: {
+    isActive: false,
+    requiredFriendTag: NaN,
```
P2: `NaN`은 계산 실패의 느낌이 있어서 초기 상태 표현에는 `null`이 더 적절할 것 같아요!

kyeoungwoon replied:
그럼 `requiredFriendTag`의 type를 `Number | null` 로 바꿔야 하는데 value에서 받을때나 등등 type 관련해서 부가적으로 처리할게 너무 많아서 `NaN` 처리한건데, (NaN은 Number type이다보니) `null`이 나을까요?

S-Gihun replied:
P3: `NaN` 은 여진님이 말씀하신 것처럼,  숫자 타입의 `계산 오류`를 의미한다고합니다. 즉, 아직 값이 없음이 아니라 `값이 있지만 잘못됨`이라는 의미에 가깝기 때문에 현재 상황에서는 `null` 이 더 적절할 거 같습니다.

kyeoungwoon replied:
null로 반영해서 변경처리 하였습니다.

S-Gihun commented on src/common/components/inputs/InputChips/InputChips.client.tsx:
```diff
@@ -0,0 +1,104 @@
+    <div
+      className={clsx(
+        'py-6pxr px-12pxr text-b3-rg rounded-4pxr flex w-full items-center gap-x-1 overflow-x-auto border text-black',
+        isFocused ? 'border-stroke-selected border-2' : 'border-gray-400',
```
P5: clsx 저희 객체 형태로 쓰기로 했던 걸로 기억하는데 혹여나 맞다면 일관성을 위해
```ts
className={clsx(
  'py-6pxr px-12pxr text-b3-rg rounded-4pxr flex w-full items-center gap-x-1 overflow-x-auto border text-black',
  {
    'border-stroke-selected border-2': isFocused,
    'border-gray-400': !isFocused,
  },
  className,
)}
```
식으로 통일하면 좋을 거 같습니다.

kyeoungwoon replied:
전부 객체로 쓰는 방법이랑 섞어 쓰는 방법 등등 clsx에서 너무 자유성을 줘서 .. 조건별로 컨벤션 정하기가 힘들 것 같은데 팀 의견을 한번 물어봐야 할 것 같긴 합니다.

duwlsssss replied:
저는 객체 형태로 작성 중이었습니다!

S-Gihun commented on src/common/components/modals/CreateNewEventModal/CreateNewEventModal.client.tsx:
```diff
@@ -0,0 +1,192 @@
+    reset,
+  } = useCreateEventConditions();
+
+  const handleRequiredFriendTagChange = (value: string) => {
+    const numberValue = Number(value);
+    setFriendTagRequirement(value && !isNaN(numberValue) ? numberValue : NaN);
+  };
```
P3: 아래 comment 에서 말씀드린, NaN -> null 변동 기준입니다.
```ts
const handleRequiredFriendTagChange = (value: string) => {
  const numberValue = Number(value);
  setFriendTagRequirement(value && !isNaN(numberValue) ? numberValue : null);
};
```
단, 이 경우 " "처럼 비어 있지는 않지만 숫자가 아닌 입력은 Number(" ") → 0으로 처리되어, 의도치 않게 0이 저장될 수 있습니다.

P5: 따라서 공백 입력 등을 명확히 걸러내기 위해 trim()과 numberValue > 0 조건을 추가한 버전입니다.
```ts
const handleRequiredFriendTagChange = (value: string) => {
  const trimmed = value.trim();
  const numberValue = Number(trimmed);
  setFriendTagRequirement(
    trimmed && !isNaN(numberValue) && numberValue > 0 ? numberValue : null
  );
};
```
`isNaN("")` 은 false 로 취급됩니다.


P5: 찾아본 결과 화면설계서 3.1에 해당 모달부분이 추가 되지 않아, 불분명하지만 만약 숫자 0 도 가능하다고 한다면 ( 태그 숫자 0명 )
```ts
setFriendTagRequirement(
  trimmed !== '' && !isNaN(numberValue) ? numberValue : null
);
```

`> 0 비교문` 을 삭제하고 빈문자열인지 확인해주는 방식도 있을 거 같습니다.

trim 을 통해 "   3    " || "     " 과 같은 것들을 막을 수 있습니다.

kyeoungwoon replied:
입력했다가 지울 때 공백이 되며 placeholder가 표시되는 것이 아닌 0이 나오는 문제를 해결하기 위한 코드였는데,
말씀하신 방법 시도해보고 변경 가능하다면 변경해두도록 하겠습니다.

kyeoungwoon replied:
반영했습니다.

S-Gihun commented on src/common/components/modals/CreateNewEventModal/CreateNewEventModal.client.tsx:
```diff
@@ -0,0 +1,192 @@
+                isNaN(conditions.friendTag.requiredFriendTag)
+                  ? ''
+                  : conditions.friendTag.requiredFriendTag.toString()
+              }
```
P3: 아래 comment 에서 말씀드린, NaN -> null 변동 기준입니다.
```ts
value={
  conditions.friendTag.requiredFriendTag === null
    ? ''
    : conditions.friendTag.requiredFriendTag.toString()
}
```

kyeoungwoon committed:
♻️ Refactor: IndividualIcon에 pxr 단위 적용

kyeoungwoon committed:
✨ Feat: TimePicker에서 특정 시간 이전을 비활성화할 수 있도록 했습니다

kyeoungwoon committed:
🔀 Merge: lefthook 파일 수정사항 반영

kyeoungwoon committed:
🩹 Chore: Image에 alt 추가

kyeoungwoon committed:
✨ Feat: DateTimePicker에 아이콘 추가

kyeoungwoon committed:
♻️ Refactor: || 연산자 대신 ?? 연산자 사용하도록 변경

kyeoungwoon committed:
♻️ Refactor: requiredFriendTag의 Type을 number | null 로 변경

기존 number만을 강제해서 NaN을 활용하였는데, 이는 값이 비어있는 것이 아닌 계산 오류 등을 나타내는 것으로 오해가 가능하여 이를 수정함

kyeoungwoon committed:
♻️ Refactor: disable 변수명 변경

kyeoungwoon committed:
♻️ Refactor: DatePicker 컴포넌트 내 util 함수 분리

kyeoungwoon committed:
✨ Feat: 참여 기간 토글 시 선택된 날짜를 삭제하도록 변경

kyeoungwoon committed:
🔀 Merge: PR #156 feature/118-input-survey

SelectBoxOption에 대한 이름 변경 적용

kyeoungwoon committed:
🔧 Fix: 버튼 잔여물 제거

kyeoungwoon committed:
♻️ Refactor: SelectBoxOption에서 UI 꺠지던 오류 해결

kyeoungwoon committed:
✨ Feat: TextBoxFieldModal애서 className을 받을 수 있도록 수정

kyeoungwoon committed:
♻️ Refactor: DateTimePicker에서 위에 제목을 받을 수 있도록 수정

kyeoungwoon committed:
✨ Feat: 새로운 팀 분위기 설문 모달 생성

스토리북은 GPT가 이상하게 만들어줘서 추후 수정 예정

kyeoungwoon committed:
♻️ Refactor: SelectBoxOption에서 아래 화살표 relative 제거

kyeoungwoon committed:
📝 Docs: 주석 추가

kyeoungwoon committed:
✨ Feat: next/image에서 링크 사용을 위해 domain 설정

kyeoungwoon committed:
♻️ Refactor: 컴포넌트 중에서 className을 props를 받지 않는 경우에 추가

kyeoungwoon committed:
♻️ Refactor: CommonText type 추가

색깔이 다른 경우가 있습니다

kyeoungwoon committed:
✨ Feat: 모달 (프로필 설정) 에서 좌측 메뉴 부분 컴포넌트화

kyeoungwoon committed:
✨ Feat: 모달 (프로필 설정) 뼈대

kyeoungwoon committed:
✨ Feat: 워크스페이스 팀원 카드 컴포넌트화

kyeoungwoon committed:
✨ Feat: 모달 (프로필 설정) - 워크스페이스 설정 부분 컴포넌트 작성

kyeoungwoon committed:
💄 Ui: 적용된 타이포그래피 변경

kyeoungwoon committed:
♻️ Refactor: 컴포넌트명 변경

명확성을 위해 ...

kyeoungwoon committed:
✨ Feat: 로그아웃 메뉴 별도로 처리하도록 분리 & 구조 개선

S-Gihun committed:
🐛 Bug: DateTimePicker & DatePickerBody.stories undefiend 문제 해결

S-Gihun committed:
🐛 Bug: 두 번 클릭시 해제 안 되는 문제 해결

strfunctionk reviewed (approved):
P5: 고생하셨습니다

S-Gihun reviewed (approved):
P5: 고생하셨습니다. DateTimePicker undefined 및 DatePickerBody undefined 이슈 해결해두었습니다.

kyeoungwoon committed:
♻️ Refactor: hook convention 적용

강하게 결합된 hook 등은 컴포넌트 폴더와 동일한 수준에 위치
